src/audio_generic.cpp

```diff
@@ -34,6 +34,10 @@ std::vector<float> GenericAudio::mixer_buffer;
 
 GenericAudio::GenericAudio() {
 	for (auto& BGM_Channel : BGM_Channels) {
+		if (BGM_Channel.midiout) {
+			BGM_Channel.midiout->Reset();
+		}
+		BGM_Channel.midiout.reset();
```

Comment:
Can you call `Reset()` in the midiout destructor? Then we don't need this messy cleanup code.